src/analytics/analytics.php

```diff
@@ -105,6 +102,34 @@ private static function is_wpvip_site() {
 			&& defined( 'FILES_CLIENT_SITE_ID' )
 			&& function_exists( '\Automattic\VIP\Stats\send_pixel' );
 	}
+
+	/**
+	 * Check if the current request should be sampled for analytics.
+	 * 
+	 * Current sampling algorithm is that every 10s or 10m, we send analytics.
+	 * 
+	 * Max calls possible based on 1 call every 10s: 8640.
+	 * Max calls possible based on 1 call every 10m/600s: 144.
+	 * So total calls possible every 10s/10m is: 8784.
+	 *
+	 * @return boolean true, if it should be sampled or false otherwise.
+	 */
+	private static function is_it_sampling_time() {
+		$current_timestamp = current_datetime();
+
+		// Get the minutes from the date.
+		$minutes = $current_timestamp->format( 'i' );
+
+		// Get the seconds from the date.
+		$seconds = $current_timestamp->format( 's' );
```

Comment:
[`format()` produces strings](https://www.php.net/manual/en/datetime.format.php), so we should probably `intval()` these or expressions like `0 !== $seconds` (i.e. `0 !== "0"`) won't work as expected.

Reply:
Actually, thinking about this a bit more we may not need checks if `seconds`/`minutes` is `0`. If I'm doing my math right, the code (with proper types) would be skipping 1/6th of the sampling times:

```
00:00:00 -> Skip, seconds is 0
00:00:10 -> Send analytics
00:00:20 -> Send analytics
00:00:30 -> Send analytics
00:00:40 -> Send analytics
00:00:50 -> Send analytics
00:01:00 -> Skip, seconds is 0
```

Reply:
Yip, updated that!